src/movies.js

```diff
@@ -1,25 +1,89 @@
 // Iteration 1: All directors? - Get the array of all directors.
 // _Bonus_: It seems some of the directors had directed multiple movies so they will pop up multiple times in the array of directors.
 // How could you "clean" a bit this array and make it unified (without duplicates)?
-function getAllDirectors(moviesArray) {}
+function getAllDirectors(moviesArray) {
+    return moviesArray.map(movie => movie.director)
+}
 
 // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct?
-function howManyMovies(moviesArray) {}
+function howManyMovies(moviesArray) {
+    let count = 0
+    moviesArray.filter(function (movie) {
+        if (movie.director === 'Steven Spielberg' && movie.genre.includes('Drama')) {
+            count++
+        }
+    })
+    return count
+}
```

Comment:
mmmmm dentro de los arrays tenemos un atributo que nos dice la cantidad de elementos que tiene verdad que es el length, entnoces podemos usarlo directamente! :D

```suggestion
function howManyMovies(moviesArray) {
    return moviesArray.filter((movie) => movie.director === 'Steven Spielberg' && movie.genre.includes('Drama')).length
}
```

Reply:
ostia claro!!!